src/types/bundle/serialized.ts

```diff
@@ -0,0 +1,95 @@
+// Serialized form of the verificationData in the Sigstore Bundle
+type SerializedVerificationData = {
+  tlogEntries: {
+    logIndex: string;
+    logId: {
+      keyId: string;
+    };
+    kindVersion:
+      | {
+          kind: string;
+          version: string;
+        }
+      | undefined;
+    integratedTime: string;
+    inclusionPromise: {
+      signedEntryTimestamp: string;
+    };
+    inclusionProof:
+      | {
+          logIndex: string;
+          rootHash: string;
+          treeSize: string;
+          hashes: string[];
+          checkpoint: { envelope: string };
+        }
+      | undefined;
+  }[];
+  timestampVerificationData: {
+    rfc3161Timestamps: { signedTimestamp: string }[];
+  };
+};
+
+// Serialized form of the x509CertificateChain option in the Sigstore Bundle's
+// verificationMaterial
+type Serializedx509CertificateChain = {
+  x509CertificateChain: {
+    certificates: { derBytes: string }[];
+  };
+};
+
+// Serialized form of the publicKey option in the Sigstore Bundle's
+// verificationMaterial
+type SerializedPublicKey = { publicKey: { hint: string } };
+
+// Serialized form of the messageSignature option in the Sigstore Bundle
+type SerializedMessageSignature = {
+  messageSignature: {
+    messageDigest:
+      | {
+          algorithm: string;
+          digest: string;
+        }
+      | undefined;
+    signature: string;
+  };
+};
+
+// Serialized form of the dsseEnvelope option in the Sigstore Bundle
+type SerializedDSSEEnvelope = {
+  dsseEnvelope: {
+    payload: string;
+    payloadType: string;
+    signatures: {
+      sig: string;
+      keyid: string;
+    }[];
+  };
+};
+
+// Serialized form of the Sigstore Bundle union type with all possible options
+// represented
+export type SerializedBundle = {
+  mediaType: string;
+  verificationData: SerializedVerificationData | undefined;
+  verificationMaterial:
+    | Serializedx509CertificateChain
+    | SerializedPublicKey
+    | undefined;
+} & (SerializedMessageSignature | SerializedDSSEEnvelope);
```

Comment:
I had a poke around and found this pattern for [creating a OneOf type](https://www.typescriptlang.org/play?#code/C4TwDgpgBAaghgGwK4QPIDMA8qBGArAPigF4pc8BtAawhAHt0z8BdAbgFgAoUSMgOzR8EIbPgA0UANJQIAD2AQ+AEwDOUGvUbkipAN5RqtKAEs+UAKKyAxsiURMGhkzwTJBZgC4ofJAgRQAXygAMih9QxATM0lPZwoYwI5ucGhUAQwAIRBJWlFCEjCDDSj1WidyWLTBYTyJDSIApLkwOgAnYCgeVPSsABUdWEQUDGwerJyRfoIkri6oAGUIVuNEYwAvCCUAWQgVFTgAcwh54wO+OGAkVug9Lih7qABbXf2jgBFT3eAPO4f7gB8wr8-n9EAc2sZgAALR5eFTAZZ8A5JEEgpSfeFwhGmZHAkEBPEAqBIZQQdCmTYoh4qU7nS7XLGI3GcRpcWYpBZLFYIdabN7zebmcx8ABuEAQdF4t04fzAcBAErgSkZOKp9zlCroSt6KRVSLVUBpZwuV12Xl0hMNpz1zNRGmMysN2P1eICFDYXFZnDZyV4i2Wqw2SgyJKUCBuQJlT02Kx1kBtSSCoSqIwtUeje0Ox1pJoZnIDPKDO0zRxOxvpEDVqhUEGFYol8fz3N5Sn5grr4sllc90x9VjofHhUBwofDXn9zaDIeU4YKab+z3RcDjEC8AHI4GAwDyrBdjAOAPQi5QAOjsIpPRvhbQgJ5HM4gAGo8CoB6wxa0aQPiAAGE8ARjXMQ8WeEts3LU1zUtUDXggD4jkxSNUXuMEIWhWEoDXeYAAkAEEACYAH18IAVgANiAy17nRBDvkw-86BI-8ABkD0kH83nMHAAHctgAMQAFgEn8AEkAGYqHwsiwAOAA5Xotlk7itxEvgwAAJUfOAAFUtJ-bDuOISj0weAJgJM+4jTpSCqMwrZzG0gBhETzAyPgAAUAEUQHcg8+DWKgkAyAAtKwlAADTI9A4G4gAOXowDeABxKhZC2AApR9zGMRyDhAASRNaYKfyoPDjHMXDYrwR8+D4vjZOMESSPS2SxLIkBZJAc46DIpBmO4n9gpIziAE4RqS4xH0eDyAHZPJgR9tKoIzzL+AkWR9YxGAACnvMNbxgrMy2s64AEokPuftBw6KzcwjPbwxPQ7SxzCtL1e00ZnTA8DygWS6A6OArBeYwcHDYEHtvata1FTtIBPDVFSUFdPS4IA). Spotted in [this blog post](https://dev.to/maxime1992/implement-a-generic-oneof-type-with-typescript-22em).

Not sure if this is the best way of representing this but seems like this could mean we can get rid of `SerializedMessageSignature | SerializedDSSEEnvelope` and just expose a `SerializedBundle` interface for all inputs and outputs.